src/acat/azext_acat/custom.py

```diff
@@ -0,0 +1,537 @@
+        from azure.cli.core.aaz import AAZStrArg
+
+        args_schema = super()._build_arguments_schema(*args, **kwargs)
+        args_schema.snapshot_name._required = False
+        args_schema.snapshot_name._registered = False
+
+        args_schema.path = AAZStrArg(
+            options=["--path"],
+            arg_group="Parameters",
+            help="Path to the downloaded file",
+            required=False,
+        )
+        args_schema.name = AAZStrArg(
+            options=["--name"],
+            arg_group="Parameters",
+            help="Name of the downloaded file without postfix",
+            required=False,
+        )
+        return args_schema
+
+    def _output(self, *args, **kwargs):
+        result = self.deserialize_output(self.ctx.vars.instance, client_flatten=True)
+
+        downloadType = str(self.ctx.args["downloadType"])
+        path = self.ctx.args["path"] or os.getcwd()
+        fname = self.ctx.args["name"] or downloadType
+        fullpath = os.path.join(path, fname) + (
+            ".pdf" if downloadType == "CompliancePdfReport" else ".csv"
+        )
+
+        if downloadType == "CompliancePdfReport":
+            import urllib.request
+
+            urllib.request.urlretrieve(
+                result["compliancePdfReport"]["sasUri"], fullpath
+            )
+        elif downloadType == "ComplianceReport":
+            saveArrayAsCsv(result["complianceReport"], fullpath)
+        elif downloadType == "ResourceList":
+            saveArrayAsCsv(result["resourceList"], fullpath)
+        else:
+            raise NameError(f"Unsupported download type {downloadType}")
+        return f"File downloaded at {fullpath}"
+
+    def pre_operations(self):
+        args = self.ctx.args
+        result = GetControlAssessment(cli_ctx=self.cli_ctx)(
+            command_args={"report_name": args.report_name, "compliance_status": "all"}
+        )
+        snapshot_name = result["snapshotName"] if result is not None else None
+        if snapshot_name is None:
+            raise ValueError(
+                f"No snapshot found for report {self.ctx.args.report_name}"
+            )
+        args.snapshot_name = snapshot_name
+
+
+class GetControlAssessment(_AcatListSnapshot):
+    class GetControlAssessmentWithDupAadToken(_AcatListSnapshot.SnapshotList):
+        CLIENT_TYPE = "AcatMgmtClient"
+
+    def _execute_operations(self):
+        self.pre_operations()
+        self.GetControlAssessmentWithDupAadToken(ctx=self.ctx)()
+        self.post_operations()
+
+    @classmethod
+    def _build_arguments_schema(cls, *args, **kwargs):
+        args_schema = super()._build_arguments_schema(*args, **kwargs)
+
+        from azure.cli.core.aaz import AAZStrArg
+
+        args_schema.compliance_status = AAZStrArg(
+            options=["--compliance-status"],
+            help="Compliance status.",
+            enum={"failed": "Failed", "succeeded": "Passed", "na": "Not Applicable"},
+            default="all",
```

Comment:
I think the "all" should be added as the enum option, otherwise the command will raise error. You can have a check by run the command without --compliance-status locally.